### PR TITLE
disttask: add StatementContext result for load data

### LINKED ISSUE #43868: add StatementContext result for load data

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close #43868 

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: util/intest/common.go [hunk 1]
```diff
@@ -17,4 +17,4 @@
 package intest
 
 // InTest checks if the code is running in test.
-const InTest = false
+const InTest = true
```
Comment (hawkingrei) [inline util/intest/common.go:20]: You don't need to modify it, it will be automatically set to true during testing.

Comment (hawkingrei) [reply util/intest/common.go:20]: I think you can modify this comment.

File: tests/realtikvtest/testkit.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ import (
 
 var (
 	// WithRealTiKV is a flag identify whether tests run with real TiKV
-	WithRealTiKV = flag.Bool("with-real-tikv", false, "whether tests run with real TiKV")
```
Comment (hawkingrei) [inline tests/realtikvtest/testkit.go:46]: It will be set as true when doing the real tikv test.

File: disttask/loaddata/proto.go [hunk 3]
```diff
@@ -31,9 +32,10 @@ const (
 // TaskMeta is the task of LoadData.
 // All the field should be serializable.
 type TaskMeta struct {
-	Plan  importer.Plan
-	JobID int64
-	Stmt  string
+	Plan    importer.Plan
+	JobID   int64
+	Stmt    string
+	Metrics Metrics
```
Comment (D3Hunter) [inline disttask/loaddata/proto.go:38]: metric means time-serial data in most case, name it as `state/result/progress`?

File: disttask/loaddata/scheduler.go [hunk 4]
```diff
@@ -179,8 +189,12 @@ func init() {
 			if err := json.Unmarshal(bs, &taskMeta); err != nil {
 				return nil, err
 			}
-			logutil.BgLogger().Info("register scheduler constructor", zap.Any("taskMeta", taskMeta))
-			return &ImportScheduler{taskMeta: &taskMeta}, nil
+			logger := logutil.BgLogger().With(zap.String("component", "scheduler"), zap.String("type", proto.LoadData), zap.Int64("ID", taskMeta.JobID))
```
Comment (D3Hunter) [inline disttask/loaddata/scheduler.go:192]: use task id? job id will be deleted soon

File: disttask/loaddata/subtask_executor.go [hunk 5]
```diff
@@ -32,10 +32,11 @@ type ImportMinimalTaskExecutor struct {
 
 // Run implements the SubtaskExecutor.Run interface.
 func (e *ImportMinimalTaskExecutor) Run(ctx context.Context) error {
-	logutil.BgLogger().Info("subtask executor run", zap.Any("task", e.task))
+	logger := logutil.BgLogger().With(zap.String("component", "minimal task executor"), zap.String("type", proto.LoadData))
```
Comment (D3Hunter) [inline disttask/loaddata/subtask_executor.go:35]: task id?

File: disttask/loaddata/scheduler.go (hunk 4 above)
Comment (GMHDBJD) [reply disttask/loaddata/scheduler.go]: Currently, the task meta does not have a task id, use table id instead now.

**Review by D3Hunter — APPROVED** [review]

File: disttask/loaddata/dispatcher.go [hunk 6]
```diff
@@ -35,17 +35,18 @@ import (
 type FlowHandle struct{}
 
 // ProcessNormalFlow implements dispatcher.TaskFlowHandle interface.
-func (h *FlowHandle) ProcessNormalFlow(ctx context.Context, handle dispatcher.TaskHandle, gTask *proto.Task) ([][]byte, error) {
+func (*FlowHandle) ProcessNormalFlow(ctx context.Context, handle dispatcher.TaskHandle, gTask *proto.Task) ([][]byte, error) {
+	logger := logutil.BgLogger().With(zap.String("component", "dispatcher"), zap.String("type", gTask.Type), zap.Int64("ID", gTask.ID))
```
Comment (lance6716) [inline disttask/loaddata/dispatcher.go:39]: the logger can be moved as a member of FlowHandle

File: disttask/loaddata/job.go [hunk 7]
```diff
@@ -58,16 +61,37 @@ func (ti *DistImporter) Param() *importer.JobImportParam {
...
+
+	ti.logger.Info("finish distribute load data", zap.Any("task meta", taskMeta))
+	var (
+		numWarnings uint64
+		numRecords  uint64
+		numDeletes  uint64
+		numSkipped  uint64
+	)
+	numRecords = taskMeta.Result.ReadRowCnt
+	// todo: we don't have a strict REPLACE or IGNORE mode in physical mode, so we can't get the numDeletes/numSkipped.
```
Comment (lance6716) [inline disttask/loaddata/job.go:88]: luckily we will use INGEST INTO, no need to keep same result message as LOAD DATA

**Review by lance6716 — APPROVED** [review]

File: disttask/loaddata/dispatcher.go (hunk 6 above)
Comment (GMHDBJD) [reply disttask/loaddata/dispatcher.go:39]: now ProcessNormalFlow and ProcessErrorFlow use different flow handle instance.

**Pull request merged**